components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -1391,6 +1395,34 @@ public List<String> getOrganizationsMetaAttributes(boolean recursive, Integer li
         return organizationMetaAttributes;
     }
 
+    @Override
+    public Map<String, String> getOrganizationNamesByIds(List<String> orgIds) throws OrganizationManagementException {
+
+        String placeholders = orgIds.stream()
+                .map(id -> "?")
+                .collect(Collectors.joining(", "));
+
+        String sql = GET_ORGANIZATION_NAME_BY_ID_MAP + placeholders + ")";
+        NamedJdbcTemplate namedJdbcTemplate = Utils.getNewTemplate();
+
+        try {
+            List<Map.Entry<String, String>> idNameEntries = namedJdbcTemplate.executeQuery(
+                    sql,
+                    (resultSet, rowNumber) -> new AbstractMap.SimpleEntry<>(
+                            resultSet.getString(VIEW_ID_COLUMN),
+                            resultSet.getString(VIEW_NAME_COLUMN)),
+                    namedPreparedStatement -> {
+                        for (int i = 0; i < orgIds.size(); i++) {
+                            namedPreparedStatement.setString(i + 1, orgIds.get(i));
+                        }
+                    });
+            return idNameEntries.stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
Let's avoid the iteration in here again. We can create a map and assign the values directly to that map.
Check: https://github.com/wso2/carbon-identity-framework/blob/79a5eeb31e4c8d5284f8ebaf5518f89469f59f14/components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/dao/impl/ActionManagementDAOImpl.java#L626